test/functional/Java14andUp/src/org/openj9/test/java/lang/Test_Thread.java

```diff
@@ -0,0 +1,177 @@
+
+        /* run thread */
+        t.start();
+        t.join();
+
+        /* Verify thread is dead and has not been interrupted */
+        AssertJUnit.assertFalse(name + "thread should be dead", t.isAlive());
+        AssertJUnit.assertTrue(name + "interrupt flag should be set", t.isInterrupted()); 
+    }
+
+    /* Interrupt thread during run, verify that interrupt flag is not set when thread stops running. 
+     * isInterrupted becomes false after InterruptedException is thrown. */
+    private volatile boolean bool_interruptDuringRun = false;
+    @Test
+    public void test_interruptDuringRun() throws Throwable {
+        String name = "interruptDuringRun: ";
+        Thread t = new Thread() {
+            public void run() {
+                synchronized(this) {
+                    try {
+                        wait();
+                    } catch (InterruptedException e) {
+                        /* expected */
+                        bool_interruptDuringRun = true;
+                    }
+                }
+            }
+        };
+
+        /* run thread */
+        t.start();
+        AssertJUnit.assertFalse(name + "thread not yet interrupted during run", t.isInterrupted());
+        t.interrupt();
+        t.join();
+
+        AssertJUnit.assertTrue(name + "InterruptedException was thrown during run", bool_interruptDuringRun);
+
+        /* Verify thread is dead and has not been interrupted */
+        AssertJUnit.assertFalse(name + "thread should be dead", t.isAlive());
+        AssertJUnit.assertFalse(name + "interrupt flag should not be set", t.isInterrupted()); 
+    }
+
+    /* Interrupt thread at start and during run, verify that interrupt flag is not set when thread stops running.
+     * isInterrupted becomes false after InterruptedException is thrown. */
+    private volatile boolean bool_interruptBeforeAndDuringRun = false;
+    @Test
+    public void test_interruptBeforeAndDuringRun() throws Throwable {
+        String name = "interruptBeforeAndDuringRun: ";
+        Thread t = new Thread() {
+            public void run() {
+                synchronized(this) {
+                    try {
+                        wait();
+                    } catch (InterruptedException e) {
+                        bool_interruptBeforeAndDuringRun = true;
+                    }
+                }
+            }
+        };
+
+        /* Verify thread is dead and has not been interrupted */
+        AssertJUnit.assertFalse(name + "thread should not be alive", t.isAlive());
+        AssertJUnit.assertFalse(name + "interrupt flag should not be set", t.isInterrupted());
+
+        /* interrupt before run */
+        t.interrupt();
+
+        AssertJUnit.assertTrue(name + "thread that has not started should have interrupt flag set", t.isInterrupted());
+
+        /* run thread */
+        t.start();
+        t.interrupt();
```

Comment:
Why the second interrupt in this test?  The interrupt state should still be set from the previous interrupt call

Reply:
I think the first one can be removed. This interrupt makes the test confusing.

Reply:
I thought the point of this test was the interrupt status from before the thread starts is still relevant when running

Reply:
That case should be covered by `test_interruptedStart`